examples/extending/extending-leaflet.md

```diff
@@ -0,0 +1,445 @@
+
+	L.CustomLayer = L.Layer.extend({
+		onAdd: function(map) {
+			var pane = map.getPane(this.options.pane);
+			this._container = L.DomUtil.create(…);
+			
+			pane.appendChild(this._container);
+			
+			// Calculate initial position of container with `L.Map.latLngToLayerPoint()`, `getPixelOrigin()` and/or `getPixelBounds()`
+			
+			L.DomUtil.setPosition(this._container, point);
+			
+			// Add and position children elements if needed
+			
+			map.on('zoomend viewreset', this._update, this);
+		},
+		
+		onRemove: function(map) {
+			L.DomUtil.remote(this._container);
+			map.off('zoomend viewreset', this._update, this);
+		},
+		
+		_update: function() {
+			// Recalculate position of container
+			
+			L.DomUtil.setPosition(this._container, point);        
+
+			// Add/remove/reposition children elements if needed
+		}
+	});
+
+How to exactly position the HTML elements for a layer depends on the specifics of the layer, but this introduction should help you to read Leaflet's layer code, and create new layers.
+
+
+
+## Handlers
+
+Map handlers are a new concept in Leaflet 1.0, and their function is to process DOM events from the browser (like `click`, `dblclick` or `mousewheel`) and change the state of the map.
+
+Handlers are relatively simple: they just need a `addHooks()` method (which runs when the handler is enabled in a map) and a `removeHooks()`, which runs when the handler is disabled. A skeleton for handlers is:
+
+	L.CustomHandler = L.Handler.extend({
+		addHooks: function() {
+			L.DomEvent.on(document, 'eventname', this._doSomething, this);
+		},
+	
+		removeHooks: function() {
+			L.DomEvent.off(document, 'eventname', this._doSomething, this);
+		},
+		
+		_doSomething: function(event) { … }
+	});
+
+This can be illustrated with a simple handler to pan the map when a mobile device is tilted, through [`deviceorientation` events](https://developer.mozilla.org/en-US/docs/Web/API/Detecting_device_orientation):
+
+	L.TiltHandler = L.Handler.extend({
+		addHooks: function() {
+			L.DomEvent.on(window, 'deviceorientation', this._tilt, this);
+		},
+	
+		removeHooks: function() {
+			L.DomEvent.off(window, 'deviceorientation', this._tilt, this);
+		},
+		
+		_tilt: function(ev) {
+			// Treat Gamma angle as horizontal pan (1 degree = 1 pixel) and Beta angle as vertical pan
+			this._map.panBy( L.point( ev.gamma, ev.beta ) );
+		}
+	});
+
+The handler can be attached to the map using `map.addHandler('tilt', L.TiltHandler)` - this will store an instance of `L.TiltHandler` as `map.tilt`. However, it's more usual to attach handlers to all maps with the `addInitHook` syntax:
+
+	L.Map.addInitHook('addHandler', 'tilt', L.TiltHandler);
+
+Our handler can now be enabled by running `map.tilt.enable()` and disabled by `map.tilt.disable()`
+	
+Moreover, if the map has a property named the same as the handler, then that handler will be enabled by default if that options is `true`, so this will enable our handler by default:
```

Comment:
Perhaps better as:

> Moreover, if the map has a property named the same as the handler, then that handler will be enabled by default _if the option with the same name is set to `true`_, so this will enable our handler by default:

(emphasis highlights my suggested change)